### PR TITLE
Feature/wordcloud demodb

### PR DESCRIPTION
**Pull request by TemsyChen**
# Description

Replace with description of PR

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [x] Yes
- [ ] No
- [ ] Not necessary

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: app/api/wordcloud_cloud.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        contour_color='steelblue',
+        ).generate(long_string).to_image()
+
+    # Convert WordCloud to JSON object
+    img = BytesIO()
+    wordcloud.save(img, "PNG")
+    img.seek(0)  
+    img_b64 = base64.b64encode(img.getvalue()).decode()
+
+    return img_b64
```
Comment (jrslagle) [inline app/api/wordcloud_cloud.py]: It looks like you may need to git pull origin main while you're checked out in this branch. This file already exists on main and is just being modified

File: app/api/wordcloud_database.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+#connect to ElephantSQL-hosted PostgreSQL
+DB_NAME = os.getenv("RDS_DB_NAME", default="OOPS")
+DB_USER = os.getenv("RDS_USERNAME", default="OOPS")
+DB_PASSWORD = os.getenv("RDS_PASSWORD", default="OOPS")
+DB_HOST = os.getenv("RDS_HOSTNAME", default="OOPS")
+DB_PORT = os.getenv("RDS_PORT", default="OOPS")
+
+DATABASE_URL = 'postgresql+psycopg2://{}:{}@{}:{}/{}'.format(DB_USER,DB_PASSWORD,DB_HOST,DB_PORT,DB_NAME)
```
Comment (jrslagle) [inline app/api/wordcloud_database.py]: Oooo. Fancy!

File: app/utils/wordcloud/wordcloud_demodata_load.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    sql = f'''INSERT INTO stories (story)
+             VALUES (%s)'''
+    cur = conn.cursor()
+    cur.execute(sql, story)
+    conn.commit()
+    return cur.lastrowid
+
+def ingest_zip():
+    '''opens the zip file of the submissions'''
+    z = ZipFile(r'C:\Users\temsy\Downloads\story transcripts.zip', 'r')
```
Comment (jrslagle) [inline app/utils/wordcloud/wordcloud_demodata_load.py]: Could you turn this into a reference to a file in the repo? Or maybe to a file on the RDS?

Comment (jrslagle) [reply app/utils/wordcloud/wordcloud_demodata_load.py]: For example:
self.all_pages = pd.read_csv("./data/37_stories_meta.csv")

**Review by jrslagle — CHANGES_REQUESTED** [review]
I would change two things. Rebase with main so it's easier to see what really changed, and so you avoid merge conflicts. And remove the reference to your local file, either by adding the transcripts to the repo or by pulling them from the database.

**Review by jrslagle — DISMISSED** [review]
Love all the changes. I haven't tested locally, but scanning the code I don't see anything to fix. Great job!

File: app/api/wordcloud_database.py (hunk 2 above)
Comment (RAV10K1) [reply app/api/wordcloud_database.py]: Agree with Jimmy on this one. Fancy!

**Review by RAV10K1 — DISMISSED** [review]
Excellent job. I haven't deployed locally but based on the stakeholder presentation it works.

File: app/utils/wordcloud/wordcloud_demodata_load.py (hunk 3 above)
Comment (TemsyChen) [reply app/utils/wordcloud/wordcloud_demodata_load.py]: I will look into this

**Review by RAV10K1 — DISMISSED** [review]
Thanks for making the changes. PR approved.

**Review by jrslagle — APPROVED** [review]
Looks great!

**Review by RAV10K1 — APPROVED** [review]
Excellent. Looks great!

**Pull request merged**